### ISSUE TITLE
Passing along an environmental variable

### ISSUE THREAD

**Issue by kgoldfeld** [USCbiostats/slurmR#28]
I am trying to fit a stan model using the `cmdstanr` package. Using cmdstan requires the declaration of the path where cmdstan sits. On my local computer, this does not need to be set once cmdstan has been installed. However, on the HPC, we need to tell the nodes where to find cmdstan by issuing a command like this (which is specific to my setup):

```r
set_cmdstan_path(path = "/gpfs/share/apps/cmdstan/2.25.0")
```

In the code below, I have commented out the two places where I know it does work. However, I'd rather not have to embed this command in the code, but would rather have it in the `Slurm_lapply` call or in some system setting. Basically, I don't really want to have to worry about this when coding.

I can comment out the second command (just before the `cmdstan_model` call) if I specify the expression in my `sbatch` code:

```
Rscript --vanilla cmdstanr_test.R -e 'set_cmdstan_path(path = "/gpfs/share/apps/cmdstan/2.25.0")'
```
However, as you well know, this does not propagate to the nodes accessed as part of the job array. My attempt at using the `rscript_opt` argument did not do the trick. Any suggestions? 

```r
library(slurmR)
library(cmdstanr)
library(simstudy)
library(data.table)

s_estimate <- function(sigma, s_model, K) {
  
  fit <- s_model$sample(
    data = list(K=K, sigma = sigma),
    seed = 123,
    chains = 4,
    parallel_chains = 4,
    refresh = 500,
    iter_warmup = 250,
    iter_sampling = 500
  )
  
  return(fit)
  
}

s_extract <- function(fit) {
  
  x <- fit$draws()
  return(as.data.table(x))
  
}

iteration <- function(sigma, s_model, K) {
  
  # set_cmdstan_path(path = "/gpfs/share/apps/cmdstan/2.25.0")
  
  fit <- s_estimate(sigma, s_model, K)
  dd <- s_extract(fit)
  
  return(dd)
  
}

#---

# set_cmdstan_path(path = "/gpfs/share/apps/cmdstan/2.25.0")
mod <- cmdstan_model("/gpfs/data/troxellab/ksg/r/cmdstanr_test.stan")

job <- Slurm_lapply(
  1:10, 
  iteration, 
  s_model = mod,
  K = 4,
  njobs = 8, 
  mc.cores = 4,
  tmp_path = "/gpfs/data/troxellab/ksg/scratch",
  overwrite = TRUE,
  job_name = "i_cmd",
  sbatch_opt = list(time = "03:00:00", partition = "cpu_short"),
  export = c("s_estimate", "s_extract"),
  plan = "wait",
  rscript_opt = list('set_cmdstan_path(path = "/gpfs/share/apps/cmdstan/2.25.0")'))

job
res <- Slurm_collect(job)
```

**Comment by kgoldfeld** [comment]
There is now a cmdstan module available, so I can execute the command `module load cmdstan/2.25.0` in the initial batch file that removes the need to execute the initial `set_cmdstan_path` call. Is there are way to add a `model load` command to `Slurm_lapply` so that the internally generated batch file includes that?  By the way, here is my batch file:

```
#!/bin/bash
#SBATCH --job-name=cmdr_test
#SBATCH --partition=cpu_short
#SBATCH --time=2:00:00                            # Time limit hrs:min:sec
#SBATCH --output=cmdstanr_test.out                # Standard output and error log
#SBATCH --mail-type=END,FAIL                      # send email when job ends
#SBATCH --mail-user=keith.goldfeld@nyulangone.org

module load r/4.0.3
module load cmdstan/2.25.0
cd /gpfs/data/troxellab/ksg/r

Rscript --vanilla cmdstanr_test.R
```

**Comment by gvegayon** [comment]
The `module load` thing is something definitely worth adding. Furthermore, USC recently created a new cluster using module. The internal API does support making personalized batch and scripts, but some of that could definitely be included in the main commands as well. I think the solution you came up with first is OK, but I understand how can that be undesired. I'll add this as a feature to be included in the next iteration.

**Comment by kgoldfeld** [comment]
OK - I can live with my solution for now, but will be on the lookout for the update.

**Comment by gvegayon** [comment]
This is now closed with #30
@kgoldfeld, check the example [here](https://github.com/USCbiostats/slurmR#example-5-using-the-preamble).
